File: BloodFlow360.Domain/Entities/Hospital.cs
using BloodFlow360.Domain.Common;

namespace BloodFlow360.Domain.Entities;

public class Hospital : BaseEntity
{
    public Guid BloodBankId { get; set; }

    public BloodBank BloodBank { get; set; } = null!;

    public string Code { get; set; } = string.Empty;

    public string Name { get; set; } = string.Empty;

    public string RegistrationNumber { get; set; } = string.Empty;

    public string Email { get; set; } = string.Empty;

    public string PhoneNumber { get; set; } = string.Empty;

    public string AddressLine1 { get; set; } = string.Empty;

    public string AddressLine2 { get; set; } = string.Empty;

    public string City { get; set; } = string.Empty;

    public string District { get; set; } = string.Empty;

    public string State { get; set; } = string.Empty;

    public string Country { get; set; } = string.Empty;

    public string PostalCode { get; set; } = string.Empty;

    public string ContactPerson { get; set; } = string.Empty;

    public bool IsActive { get; set; }
}

File: BloodFlow360.React/src/types/hospital.ts
export interface Hospital {
  id: string;
  bloodBankId: string;
  bloodBankName: string;
  code: string;
  name: string;
  registrationNumber: string;
  email: string;
  phoneNumber: string;
  addressLine1: string;
  addressLine2: string;
  city: string;
  district: string;
  state: string;
  country: string;
  postalCode: string;
  contactPerson: string;
  isActive: boolean;
  createdAt: string;
}

export interface CreateHospital {
  bloodBankId: string;
  code: string;
  name: string;
  registrationNumber: string;
  email: string;
  phoneNumber: string;
  addressLine1: string;
  addressLine2: string;
  city: string;
  district: string;
  state: string;
  country: string;
  postalCode: string;
  contactPerson: string;
}

export interface UpdateHospital {
  code: string;
  name: string;
  registrationNumber: string;
  email: string;
  phoneNumber: string;
  addressLine1: string;
  addressLine2: string;
  city: string;
  district: string;
  state: string;
  country: string;
  postalCode: string;
  contactPerson: string;
  isActive: boolean;
}
